gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -373,6 +382,18 @@ public GoogleCloudStorageImpl(
             ? null
             : this.storage.getRequestFactory().getInitializer();
 
+    if (this.storageOptions.isMetricsEnabled()) {
+      checkArgument(
```

Comment:
Maybe this redundant check as case will fail anyway with quite descriptive exception?